proof/access-control/ARM/ArchInterrupt_AC.thy

```diff
@@ -62,7 +63,12 @@ crunches arch_check_irq for inv[Interrupt_AC_assms, wp]: P
 
 end
 
-requalify_consts ARM_A.arch_authorised_irq_ctl_inv
+
+context begin interpretation Arch .
+
+requalify_consts arch_authorised_irq_ctl_inv
+
+end
```

Comment:
Just curious: was there a problem with the way it was previously phrased?

IIUC, the old way should have the same effect as what you've replaced it with, but the old way is probably faster, since it doesn't require unpacking the whole `Arch` locale. Since you're in an `ARM` theory, I guess it's ok to mention `ARM_A` explicitly.

Reply:
We're probably not very consistent in the way we do this, though. I expect there are many places we unpack all  of `Arch` in an arch theory, where specific constants  or facts would work just as well.